From 26ce8fc25c002c9184afcf77dbf3bdccde3a8d4e Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Tue, 28 Apr 2020 13:59:42 +0200
Subject: [PATCH] MIPI MAXEN display GPIO configuration

The MIPI MAXEN display needs set these GPIOs to
have the correct setup

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo  <vicentiu@balena.io>
---
 board/compulab/cl-som-imx8/cl-som-imx8.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/board/compulab/cl-som-imx8/cl-som-imx8.c b/board/compulab/cl-som-imx8/cl-som-imx8.c
index 4b0c12c2d4..2a14ceb959 100644
--- a/board/compulab/cl-som-imx8/cl-som-imx8.c
+++ b/board/compulab/cl-som-imx8/cl-som-imx8.c
@@ -332,6 +332,21 @@ int board_init(void)
 
 	setup_usbmux();
 
+	gpio_request(IMX_GPIO_NR(4, 5),"tft_stdby");
+	gpio_direction_output(IMX_GPIO_NR(4, 5), 1);
+
+	gpio_request(IMX_GPIO_NR(5, 18),"tft_bcl_pwm");
+	gpio_direction_output(IMX_GPIO_NR(5, 18), 0);
+
+	gpio_request(IMX_GPIO_NR(4, 1),"tft_shlr");
+	gpio_direction_output(IMX_GPIO_NR(4, 1), 0);
+
+	gpio_request(IMX_GPIO_NR(4, 16),"tft_updw");
+	gpio_direction_output(IMX_GPIO_NR(4, 16), 1);
+
+	gpio_request(IMX_GPIO_NR(3, 25),"tft_rst");
+	gpio_direction_output(IMX_GPIO_NR(3, 25), 1);
+
 #if defined(CONFIG_USB_DWC3) || defined(CONFIG_USB_XHCI_IMX8M)
 	init_usb_clk();
 #endif
-- 
2.17.1

